tests/samplerowkeys_test.go

```diff
@@ -313,9 +313,13 @@ func TestSampleRowKeys_Retry_WithRetryInfo(t *testing.T) {
 
 	// 4b. Verify retry backoff time is correct
 	firstReq := <-recorder
-	retryReq := <-recorder
```

Comment:
This was blocking indefinitely when the client didn't retry

Reply:
Same here, I think the client should skip the test explicitly so the behavior is tested.

Reply:
As written, the test will still fail if the client isn't conformant, but instead of hanging indefinitely the test will fail explicitly fail via the line `t.Error("Timeout waiting for retry request")`.